app/models/mixins/supports_feature_mixin.rb

```diff
@@ -67,35 +102,57 @@ def unsupported
     @unsupported ||= {}
   end
 
-  class_methods do
-    def unsupported_reason(feature)
-      public_send("supports_#{feature}?") unless unsupported.key?(feature)
-      unsupported[feature]
-    end
+  def guard_queryable_feature(feature)
+    self.class.send(:guard_queryable_feature, feature)
+  end
 
+  class_methods do
+    # This is the DSL
     def supports(feature, &block)
+      guard_queryable_feature(feature)
       send(:define_supports_methods, feature, true, &block)
     end
 
     def supports_not(feature, reason: nil)
+      guard_queryable_feature(feature)
       send(:define_supports_methods, feature, false, reason)
     end
 
+    # This is for querying features
+    def supports?(feature)
+      guard_queryable_feature(feature)
+      send("supports_#{feature}?")
+    end
+
+    def unsupported_reason(feature)
+      guard_queryable_feature(feature)
+      public_send("supports_#{feature}?") unless unsupported.key?(feature)
+      unsupported[feature]
+    end
+
     private
 
+    def guard_queryable_feature(feature)
+      unless QUERYABLE_FEATURES.key?(feature.to_s.to_sym)
```

Comment:
Because `nil.to_sym` does not work, thats more a reflex of me, in this case I can skip it :+1: thanks